lib/ramble/ramble/renderer.py

```diff
@@ -67,8 +71,44 @@ def render_objects(self, variables, matrices=None):
             object_variables[name] = value
 
         new_objects = []
+        defined_zips = {}
+        consumed_zips = set()
         matrix_objects = []
 
+        if zips:
+            zipped_vars = set()
+
+            for zip_group, group_def in zips.items():
+                # Create a new defined zip
+                defined_zips[zip_group] = {'vars': {}, 'length': -1}
+                cur_zip = defined_zips[zip_group]
+                for var_name in group_def:
+                    if var_name in zipped_vars:
+                        tty.die(f'Variable {var_name} is used across multiple zips.\n'
+                                'Ensure it is only used in a single zip')
+
+                    # Ensure the variable is defined
```

Comment:
nit: it functionally doesn't matter, but it feels like "is this a real var" makes sense as the first check not down in the list? Could be worth hoisting but I don't strictly care

Reply:
Done